main/src/com/google/refine/exporters/MdExporter.java

```diff
@@ -0,0 +1,64 @@
+package com.google.refine.exporters;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.refine.browsing.Engine;
+import com.google.refine.model.Project;
+import net.steppschuh.markdowngenerator.MarkdownSerializationException;
+import net.steppschuh.markdowngenerator.table.Table;
+import net.steppschuh.markdowngenerator.link.Link;
+import net.steppschuh.markdowngenerator.text.Text;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Properties;
+
+public class MdExporter implements StreamExporter{
+    @Override
+    public String getContentType() {
+        return "text/markdown";
+    }
+
+    @Override
+    public void export(Project project, Properties params, Engine engine, OutputStream outputStream) throws IOException {
+        final Table.Builder mdDoc = new Table.Builder();
+
+        TabularSerializer serializer = new TabularSerializer() {
+            @Override
+            public void startFile(JsonNode options) {
+            }
+
+            @Override
+            public void endFile() {
+            }
+
+            @Override
+            public void addRow(List<CellData> cells, boolean isHeader) {
+                String[] cellData = new String[cells.size()];
+                for(int i = 0; i < cells.size(); i++){
+                    CellData cellDatatmp = cells.get(i);
+                    cellData[i] = "";
+                    if(cellDatatmp != null && cellDatatmp.text != null){
+                        String textInCell = cellDatatmp.text;
+                        textInCell = textInCell.replaceAll("([`|])", "\\\\$1");
```

Comment:
I am surprised that this is not handled directly by the `net.steppschuh.markdowngenerator` library: this ought to be done there natively, no? Perhaps it would be worth reporting this to them.

Reply:
This library doesn't handle the situation that `|` in the markdown table, I will report it to them (but I don't think he will handle it quickly, cause the last update was a year ago)

Reply:
I suspect we might need to escape more characters here (for instance, the ones that are used to format text as bold or italics).